tests/unit/nupic/serializable_test.py

```diff
@@ -0,0 +1,80 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2017, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import unittest
+
+import capnp
+import serializable_test_capnp
+
+from nupic.serializable import  Serializable
+
+
+
+class SerializableTest(unittest.TestCase):
+
+  def testABCProtocolEnforced(self):
+
+    class Foo(Serializable):
+      pass # read(), write(), getCapnpSchema() not implemented here
+
+    with self.assertRaises(TypeError):
+      Foo()
+
+
+  def testReadFromAndWriteToFile(self):
+    """ Test generic usage of serializable mixin class """
+
+    class Bar(object):
+      pass
+
+
+    class Foo(Bar, Serializable):
+
+
+      def __init__(self, bar):
+        self.bar = bar
+
+
+      @classmethod
+      def getCapnpSchema(cls):
+        return serializable_test_capnp.Foo
+
+
+      @classmethod
+      def read(cls, proto):
+        foo = object.__new__(cls)
+        foo.bar = proto.bar
+        return foo
+
+
+      def write(self, proto):
+        proto.bar = self.bar
+
+
+    self.addCleanup(os.remove, "foo.data")
```

Comment:
If `writeToFile` fails because of a bug and doesn't write a file then `os.remove` will also fail. I'm not sure what this will look like in the test output but it could be confusing.

It is a rare case and the errors would reflect a real bug so I don't think you should address it explicitly but thought I'd point it out in case you think there is any cause for concern.